### DISCUSSION TITLE
Whether mqtt plugin can support QOS2?  Is there a plan?   thanks.

### DISCUSSION THREAD

**Discussion by objectboy2016** [rabbitmq/rabbitmq-server#10291] [Other]
### Describe the solution you'd like

mqtt plugin message level implementation qos2

**Comment by michaelklishin** [comment]
It does not support QoS 2, `SUBSCRIBE` frames with QoS 2 are treated as if they were using QoS 1.

There are no plans to work on QoS 2 support as it would make no practical difference w.r.t. key fundamental failure scenarios or distributed system design challenges.

Somehow this is missing from the [Limitations](https://rabbitmq.com/mqtt.html#limitations) section in the docs, I'll update it now.
